packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -13,44 +14,101 @@ type DisplayContent =
   | undefined
   | unknown;
 
+type ResultFontSize = "sm" | "base" | "xs" | "lg" | "xl" | "2xl" | "3xl";
+
 export const displayTxResult = (
   displayContent: DisplayContent | DisplayContent[],
-  asText = false,
+  fontSize: ResultFontSize = "base",
 ): string | ReactElement | number => {
   if (displayContent == null) {
     return "";
   }
 
   if (typeof displayContent === "bigint") {
-    try {
-      const asNumber = Number(displayContent);
-      if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
-        return asNumber;
-      } else {
-        return "Ξ" + formatEther(displayContent);
-      }
-    } catch (e) {
-      return "Ξ" + formatEther(displayContent);
-    }
+    return <NumberDisplay value={displayContent} />;
   }
 
-  if (typeof displayContent === "string" && isAddress(displayContent)) {
-    return asText ? displayContent : <Address address={displayContent} />;
+  if (typeof displayContent === "string") {
+    if (isAddress(displayContent)) {
+      return <Address address={displayContent} size={fontSize} />;
+    }
+
+    if (isHex(displayContent)) {
+      return displayContent; // don't add quotes
+    }
   }
 
   if (Array.isArray(displayContent)) {
-    const mostReadable = (v: DisplayContent) =>
-      ["number", "boolean"].includes(typeof v) ? v : displayTxResultAsText(v);
-    const displayable = JSON.stringify(displayContent.map(mostReadable), replacer);
-
-    return asText ? (
-      displayable
-    ) : (
-      <span style={{ overflowWrap: "break-word", width: "100%" }}>{displayable.replaceAll(",", ",\n")}</span>
-    );
+    return <ArrayDisplay value={displayContent} size={fontSize} />;
+  }
+
+  if (typeof displayContent === "object") {
+    return <StructDisplay value={displayContent} size={fontSize} />;
   }
 
   return JSON.stringify(displayContent, replacer, 2);
 };
 
-const displayTxResultAsText = (displayContent: DisplayContent) => displayTxResult(displayContent, true);
+const NumberDisplay = ({ value }: { value: bigint }) => {
+  const [isEther, setIsEther] = useState(false);
+
+  const asNumber = Number(value);
+  if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
+    return String(value);
+  }
+
+  return (
+    <span>
+      {isEther ? "Ξ" + formatEther(value) : String(value)}
+      <span
+        className="tooltip tooltip-secondary font-sans ml-2"
+        data-tip={isEther ? "Multiply by 1e18" : "Divide by 1e18"}
+      >
+        <button className="btn btn-primary btn-circle btn-xs" onClick={() => setIsEther(!isEther)}>
+          <ArrowsRightLeftIcon className="h-3 w-3" />
```

Comment:
Ohh I think we decided to change this to "/" and "*" and also tooltip text right ? 

https://github.com/scaffold-eth/scaffold-eth-2/pull/853#pullrequestreview-2085318361

Reply:
Oh, I changed the tooltip and the icon is now ArrowsRightLeft as I commented: https://github.com/scaffold-eth/scaffold-eth-2/pull/853#issuecomment-2135369866

Do you think it would be better to change it to "/" and "*"?

Reply:
Umm I think it's better be explicit ? 

But cc @carletex and @rin-st once

Reply:
For me both variants are good but yes probably "/" and "*" are better.

Reply:
Same, not feeling strong, but might lean into `/` and `*`

----

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/50d8c396-c0d1-4713-9e7c-da1d13ede76e)

We are still giving the button too much importance:
- It's using the primary color. (meant for main actions)
- It's too tall: It's misaligning the key and the value too (e.g. [0] and 230000..)

I think it should be waaaay more subtle. This feature is nice to have, but the UI for this shouldn't draw attention. 

Tried without a background: maybe it's too subtle? haha

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/a7e5eb36-011f-4631-a3b0-838fffa760b1)

Another option would be to click the number it self (with a tooltip when hovering). I don't like this one too much because i use double click for selecting the text/word and copy.

We can also ask Andrea, so she could give us a designer's take.

Reply:
> We are still giving the button too much importance:

yeah makes sense, updated it to be btn-ghost and decreased the opacity to 65, this is how it looks now : 

| normal | on hover | 
| ------- | ----- | 
| ![Screenshot 2024-06-10 at 8 32 45 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/abdca4e6-b03f-492c-8d57-5e499a0f1d2a) | ![Screenshot 2024-06-10 at 8 32 52 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/cddc7124-aae1-451b-bd67-f74e258f0fa9) | 

What do you think ? But yeah we can ask Andrea to find better opotion

---

Umm also  kept the same ArrowLeftRightIcon instead of `/` and `*` since they  were not looking that great : 

| `/` | `*` | 
| ------- | ----- | 
| ![Screenshot 2024-06-10 at 8 08 08 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/c9e2645e-3385-4c49-8504-37829df1fe6c) | ![Screenshot 2024-06-10 at 8 08 30 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/58b33422-66e8-446c-abed-b6531213c711) |

Reply:
That looks better I think! The only thing is that the line is not aligned (because of the button height?)

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/05bd29cc-1685-4a60-a11f-895f2073eed8)

Reply:
Ohh yes yes just fixed it at https://github.com/scaffold-eth/scaffold-eth-2/pull/853/commits/6d168a236bf41a57ca79c88662b65dac7aa53059: 

| Before | After | 
| ------- | ----- | 
| ![Screenshot 2024-06-10 at 9 26 01 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/9fa94598-1ae3-4e23-a9e6-65573a00c5cc) | ![Screenshot 2024-06-10 at 9 25 46 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/23e91e4a-0c38-4694-aff7-5678d53fa19d) |